Marlin/src/lcd/ultralcd.cpp

```diff
@@ -2743,48 +2743,54 @@ void kill_screen(const char* lcd_msg) {
     void _goto_tower_z() { _man_probe_pt(cos(RADIANS( 90)) * delta_calibration_radius, sin(RADIANS( 90)) * delta_calibration_radius); }
     void _goto_center()  { _man_probe_pt(0,0); }
 
-    void _lcd_set_delta_height() {
-      update_software_endstops(Z_AXIS);
+  #endif // DELTA_CALIBRATION_MENU
+
+  #if ENABLED(DELTA_CALIBRATION_MENU) || ENABLED(DELTA_AUTO_CALIBRATION)
+
+    void lcd_recalc_delta_settings() {
+      recalc_delta_settings(delta_radius, delta_diagonal_rod, delta_tower_angle_trim);
```

Comment:
`recalc_delta_settings` doesn't call `update_software_endstops`, so the endstops won't be updated if the `delta_height` is adjusted.

Reply:
I'll better include it then in recalc_delta_settings for good measure; but after every recalc_delta_settings homing is already required.

Reply:
Would it be a good idea to reset `axis_is_homed` in `recalc_delta_settings` to force the machine to be homed before continuing?

Reply:
Was already included:

axis_homed[X_AXIS] = axis_homed[Y_AXIS] = axis_homed[Z_AXIS] = false;

Reply:
Whenever any change occurs in the values that apply to the kinematics, the customary solution has been to get the current values from the steppers, convert those back into cartesian coordinates, and update the `current_position` so that it matches the physical position of the machine.

In other words…
```cpp
  set_current_from_steppers_for_axis(ALL_AXES);
  SYNC_PLAN_POSITION_KINEMATIC();
```

Reply:
The re-conversion from cartesian back into steps via `SYNC_PLAN_POSITION_KINEMATIC` is needed to mitigate any potential `float` imprecision.